test/dynamo/test_profiler.py

```diff
@@ -192,6 +192,47 @@ def fn(x, y):
             ],
         )
 
+    def test_profiler_enabled(self):
+        def fn(x):
+            x = torch.sin(x)
+            if torch.autograd._profiler_enabled():
+                return torch.cos(x)
+            else:
+                return torch.sigmoid(x)
+
+        opt_fn = torch.compile(fn, backend="eager", fullgraph=True)
+        x = torch.randn(4)
+
+        ref = fn(x)
+        res = opt_fn(x)
+        self.assertEqual(ref, res)
+
+        with torch.autograd.profiler.profile():
+            ref = fn(x)
+            res = opt_fn(x)
+            self.assertEqual(ref, res)
+
+    def test_profiler_record_function_ignore(self):
+        def fn(x):
+            x = torch.sin(x)
+            if torch.autograd._profiler_enabled():
+                with torch.autograd.profiler.record_function("dummy"):
```

Comment:
Just making sure: do we make it clear that profiler is ignored by compile?

Reply:
This is pre-existing. There was no warning or anything. But perhaps we should make a warning.